### ISSUE TITLE
Resumed app load rewardedvideo error

### ISSUE THREAD

**Issue by huachang** [mopub/mopub-android-sdk#342]
MoPubRewardedVideos.hasRewardedVideo() API return false while MoPubRewardedVideos.loadRewardedVideo() API do not load as it logged "Did not queue rewarded ad request for ad unit XXX. This ad unit already finished loading and is ready to show"
The possible reason is : hasRewardedVideo() use MoPubRewardedVideoManager.isPlayable() while loadRewardedVideo() use rewardedAdsLoaders.canPlay(). And the former needs not only rewardedAdsLoaders.canPlay() to be true but also customEvent.isReady(). 

- [ ] I am submitting a bug report for existing functionality
- [ ] I visited https://developers.mopub.com/ and found no answer
- [ ] I checked https://twittercommunity.com/c/advertiser-api/mopub to make sure that this issue has not already been filed
- [ ] I checked to make sure that this issue has not already been filed

#### Steps to reproduce the behavior:
Please list all relevant steps to reproduce the observed behavior.

#### Expected behavior:
As concisely as possible, describe the expected behavior.

#### Observed behavior:
As concisely as possible, describe the observed behavior.

#### Evidence:
Device log files, Network log file, etc.

**Comment by chauduyphanvu** [comment]
@huachang, do you have reliable steps that we can use to reproduce that error? Are you requesting videos sequentially or concurrently? If possible, please include some code snippets.

Vu Chau
MoPub SDK Team

**Comment by huachang** [comment]
It happens when reopen the app using MoPub SDK after sometime like an hour or half. 

> @huachang, do you have reliable steps that we can use to reproduce that error? Are you requesting videos sequentially or concurrently? If possible, please include some code snippets.
> 
> Vu Chau
> MoPub SDK Team

**Comment by chauduyphanvu** [comment]
When you say "reopen", what exactly does that mean? Do you background the app and then bring it to the foreground? Or do you already cache an ad but make a new ad request an hour later? I'm looking for a reliable sequence of steps so we can reproduce and fix the issue.

Could you also confirm if you are able to reproduce that issue in the MoPub Android sample app on this repository? Thank you.

**Comment by huachang** [comment]
I mean background the app and bring it to foreground after an hour. In this case, when i try to load，mopub tell that "Did not quene the request .....". Because it detected that already have an ad using canplay api.  While i use haverewardedvideo api to query it returns false.    So I searched the source code. and find that haverewardedvideo api called isplayable not canplay. -------- 原始邮件 --------主题：Re: [mopub/mopub-android-sdk] Resumed app load rewardedvideo error (#342)发件人：Vu 收件人：mopub/mopub-android-sdk 抄送：huachang ,State change When you say "reopen", what exactly does that mean? Do you background the app and then bring it to the foreground? Or do you already cache an ad but make a new ad request an hour later? I'm looking for a reliable sequence of steps so we can reproduce and fix the issue.
Could you also confirm if you are able to reproduce that issue in the MoPub Android sample app on this repository? Thank you.

**Comment by chauduyphanvu** [comment]
@huachang, thank you for the confirmation. I understand the use case, but I'm going to need more details. When you bring the app to the background, have you played the ad that's in the cache yet? Or is the ad not played and you make a new ad request when the app is foregrounded? If the latter, the SDK does not allow for that to happen unless your new ad request uses a different ad unit ID.

**Comment by huachang** [comment]
I reproduced the error by click the home buttom to bring my app to background when there was a rewardedvideo was showing. And then I clicked my app icon to bring my app to foreground. I fixed this problem by replace if (sInstance.rewardedAdsLoaders.canPlay(adUnitId))toif(hasVideo(adUnitId))in loadVideo of MoPubRewaidedVideoManager.It works. But I guess that this solution will reduce the shown/request ratio of rewardedvideo ads.I'm looking forward to an official solution.-------- 原始邮件 --------主题：Re: [mopub/mopub-android-sdk] Resumed app load rewardedvideo error (#342)发件人：Vu 收件人：mopub/mopub-android-sdk 抄送：huachang ,Mention @huachang, thank you for the confirmation. I understand the use case, but I'm going to need more details. When you bring the app to the background, have you played the ad that's in the cache yet? Or is the ad not played and you make a new ad request when the app is foregrounded? If the latter, the SDK does not allow for that to happen unless your new ad request uses a different ad unit ID.
